extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java

```diff
@@ -50,7 +50,7 @@
  * Class that is responsible for running the HTTP based authentication
  */
 @Singleton
-public class HttpAuthenticator {
+public final class HttpAuthenticator {
```

Comment:
Any reason? That would avoid users to reuse parts of the logic.

Reply:
> Any reason? 

TL;DR; when it is not final, users can start reusing it, but we change it. Long version (you don't need to read it):

- It's runtime class that uses constructor for CDI injection.
- We changed this class many times during last few releases
- I thought setting `final` gives clear message that it should not be extended
- recently we had user that reported change of class in OIDC in runtime module, before we had user that reported that I removed some method (I think it was on this very class)

> That would avoid users to reuse parts of the logic.

It's fine if we expect it, but if with every change we need to consider users that may have used this class, it means we cannot do whatever in please in runtime module.

Whatever I add, I try to make clear line (usually by visibility or by `sealed`) what I expect users to use and what I don't expect, so that users are not surprised when it is changed.

But I definitely don't need to do changes like this to existing classes, so if you want, I'll just drop it.

Reply:
Ok, so it would need to be added in the migration guide (usages of these classes were not expected, and these classes should be considered only for internal use)

Reply:
Agreed, but I cannot adjust migration guide myself. @sberyozkin please either add note to the migration guide or I could drop this change. I make these sneaky visibility / `final` changes quite often, so you probably didn't realize I changed it :-)